### PR TITLE
Promote alto auth codecs to stable [API-2134]

### PR DESCRIPTION
**Pull request by srknzl**
As part of promiting ALTO to stable in 5.4, we will get rid of experimental authentication codecs for Alto. This PR does that. 

Fixes API-2134

protocol PR : https://github.com/hazelcast/hazelcast-client-protocol/pull/480

Breaking changes (list specific methods/types/messages):
* None because we are removing some experimental codec

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible
- [x] Send backports/forwardports if fix needs to be applied to past/future releases
- [x] New public APIs have `@Nonnull/@Nullable` annotations
- [x] New public APIs have `@since` tags in Javadoc

### CODE REVIEW

**Comment by srknzl** [comment]
run-lab-run

**Review by ihsandemir — APPROVED** [review]

File: docs/design/alto/client-interaction.md [hunk 1]
```diff
@@ -16,7 +16,8 @@ Clients will continue to use the classic Hazelcast ports for the
 initial authentication.
 
 If the TPC is enabled on the client-side, clients will make use of the two 
-newly added codecs to authenticate with the cluster.
+newly added codecs to authenticate with the cluster. (note: these codecs were removed later on and 
+`tpcPorts` and `tpcToken` fields were added to the stabled codecs. This was done as part of making Alto stable)
```
Comment (vbekiaris) [inline docs/design/alto/client-interaction.md:20]: minor typo?
```suggestion
`tpcPorts` and `tpcToken` fields were added to the stable codecs. This was done as part of making Alto stable)
```

**Review by vbekiaris — APPROVED** [review]

File: docs/design/alto/client-interaction.md (hunk 1 above)
Comment (srknzl) [reply docs/design/alto/client-interaction.md]: Thanks, handled.

File: docs/design/alto/client-interaction.md [hunk 2]
```diff
@@ -67,7 +68,8 @@ classic ports.
 First, 3-bytes long protocol identifier is sent. (bytearray representation of `CP2`).
 
 Then, the client authenticates to the TPC event loop using the new 
-`ExperimentalTpcAuthenticationCodec`, which has the following definition.
+`ExperimentalTpcAuthenticationCodec`, which has the following definition. (note: this codec was later renamed to
```
[the remark this answers is not in the document]
Comment (srknzl) [reply docs/design/alto/client-interaction.md]: Thanks, handled.

**Review by pveentjer — APPROVED** [review]

**Pull request merged**